vcr/verifier/verifier.go

```diff
@@ -136,10 +151,36 @@ func (v *verifier) Validate(credentialToVerify vc.VerifiableCredential, at *time
 		return fmt.Errorf("unable to resolve valid signing key at given time: %w", err)
 	}
 
-	// Try first with the correct LDProof implementation
 	return ldProof.Verify(signedDocument.DocumentWithoutProof(), signature.JSONWebSignature2020{ContextLoader: v.jsonldManager.DocumentLoader()}, pk)
 }
 
+func (v *verifier) validateJWTCredential(credential vc.VerifiableCredential, at *time.Time) error {
+	var keyID string
+	_, err := crypto.ParseJWT(credential.Raw(), func(kid string) (crypt.PublicKey, error) {
+		keyID = kid
+		return v.resolveSigningKey(kid, credential.Issuer.String(), at)
+	})
+	if err != nil {
+		return fmt.Errorf("unable to validate JWT credential: %w", err)
+	}
+	if keyID != "" && strings.Split(keyID, "#")[0] != credential.Issuer.String() {
+		return errVerificationMethodNotOfIssuer
+	}
```

Comment:
Is this the only validation that needs to be done according to spec?

Reply:
The spec does not even state this (it vaguely hints it could do something like this), but we do it for JSON-LD, do we should do the same for JWT to avoid confusion (leading to vulnerabilities)